### PR TITLE
fix: Use real name of the tag if it is unknwon

### PR DESCRIPTION
**Pull request by zxdxjtu**
When generating LSIF file using spoon, we use (CtClass)ctClass.getDocComment to fetch doc comment of the class to display hover result.
However, spoon regards user-defined tag (e.g. @email @create ) as '@unknown', it is really unfriendly for printing hover content. Usually the class hover would be:
    Example class
    @author zxdxjtu
    @unknwon zhangxindong888@163.com
    @unknown 2020-07-30 10:24

To solve this problem, I retain the real name of the comment tag and use real name of the rag instead of TagType.name when printing doc comment to avoid '@unknown' condition.

### CODE REVIEW

File: src/main/java/spoon/javadoc/internal/JavadocBlockTag.java [hunk 1]
```diff
@@ -32,20 +32,22 @@
 	private JavadocDescription content;
 	private String name = "";
 	private String tagName;
+	private String tagRealName;
 
-	public JavadocBlockTag(CtJavaDocTag.TagType type, String content) {
+	public JavadocBlockTag(CtJavaDocTag.TagType type, String tagRealName, String content) {
```
Comment (monperrus) [inline src/main/java/spoon/javadoc/internal/JavadocBlockTag.java:37]: please avoid breaking changes, add a constructor instead

File: src/main/java/spoon/reflect/factory/CodeFactory.java [hunk 2]
```diff
@@ -710,7 +710,7 @@ public CtComment createInlineComment(String content) {
 	 * @param type The tag type
 	 * @return a new CtJavaDocTag
 	 */
-	public CtJavaDocTag createJavaDocTag(String content, CtJavaDocTag.TagType type) {
+	public CtJavaDocTag createJavaDocTag(String content, CtJavaDocTag.TagType type, String realName) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CodeFactory.java:713]: no breaking change in the factory

**Comment by zxdxjtu** [comment]
Forgive me for mistakes on code modification. Recently I don't have much time on fixing failed test cases, I will perfect it as soon as possible.

File: src/test/java/spoon/processing/CtGenerationTest.java [hunk 3]
```diff
@@ -151,7 +151,7 @@ public void testGenerateCloneVisitor() {
 		CtClass<Object> expected = build(new File("./src/main/java/spoon/support/visitor/clone/CloneBuilder.java")).Class().get("spoon.support.visitor.clone.CloneBuilder");
 		assertThat(actual).isEqualTo(expected);
 
-		// cp ./target/generated/spoon/support/visitor/clone/CloneVisitor.java  ./src/main/java/spoon/support/visitor/clone/CloneVisitor.java
```
Comment (monperrus) [inline src/test/java/spoon/processing/CtGenerationTest.java:154]: undo

**Comment by monperrus** [comment]
Hi @zxdxjtu Thanks for the ping. I don't see new changes in the diff since my previous review: https://github.com/INRIA/spoon/pull/3513#pullrequestreview-468230791

Any idea?

**Comment by monperrus** [comment]
FYI, this stackoverflow question seems about this: https://stackoverflow.com/questions/63810364/getting-text-of-non-javadoc-tag-inside-doc-comment

**Comment by monperrus** [comment]
Hi @zxdxjtu we're almost there, one more change and failing tests to fox :)

Then we merge!

**Comment by zxdxjtu** [comment]
Hi, @monperrus , it seems that I have already fixed all travis issues before, but 'travis failed' appeared again when I merged master. Travis said multiple ' duplicate classes found', I am not so familiar with travis, could you help me find the way to fix travis issue?

**Review by monperrus — APPROVED** [review]
Fixed CI by pushing to your branch. LGTM, will merge.

**Comment by monperrus** [comment]
Thanks a lot for your contribution!

**Pull request merged**